components/common/hocs/cloud/with_get_cloud_subcription.tsx

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ComponentType} from 'react';
+import PropTypes from 'prop-types';
+
+import {isEmpty} from 'lodash';
+
+import {Subscription} from 'mattermost-redux/types/cloud';
+
+interface Actions {
+    getCloudSubscription: () => void;
+}
+
+interface UsedHocProps {
+    subscription: Subscription;
+    isCloud: boolean;
+    actions: Actions;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function withGetCloudSubscription<P>(WrappedComponent: ComponentType<P>): any {
+    return class extends React.Component<P & UsedHocProps> {
+        static propTypes = {
```

Comment:
We shouldn't be using `PropTypes` with TypeScript. TypeScript does its job without it.